cmd/main_test.go

```diff
@@ -25,11 +26,35 @@ func TestHelloWorld(t *testing.T) {
 	assert.NoError(t, err)
 	assert.Equal(t, 200, res.StatusCode)
 
-	body, err := ioutil.ReadAll(res.Body)
+	body, err := io.ReadAll(res.Body)
 	assert.NoError(t, err)
 	assert.Equal(t, "Hello world", string(body))
 }
 
+func TestTodoHandler(t *testing.T) {
+	r := chi.NewRouter()
+	r.Use(middleware.Logger)
+	r.Post("/todo", func(w http.ResponseWriter, r *http.Request) {
```

Comment:
You're not testing your server here man, create an instance of your server struct, register your routes then pass the handler to hettptest server. Then I'll teach you how to debug handlers